tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs

```diff
@@ -2,84 +2,157 @@
+
+    private readonly string _unknownText = CommonHelper.GetLocalizedString("UnknownText");
+
+    private const string UwpCommandLinePattern = @"^(.*) -ServerName:(.*?)\.AppX(.*?)\.mca$";
+
+    [GeneratedRegex(UwpCommandLinePattern)]
+    private static partial Regex UwpCommandLineRegex();
+
+    [ObservableProperty]
+    private int _processId = 0;
+
+    [ObservableProperty]
+    private int _basePriority = 0;
+
     [ObservableProperty]
-    private int processId = 0;
+    private int _priorityClass = 0;
 
     [ObservableProperty]
-    private int basePriority = 0;
+    private string _mainModuleFileName = string.Empty;
 
     [ObservableProperty]
-    private int priorityClass = 0;
+    private string _cpuArchitecture = string.Empty;
 
     [ObservableProperty]
-    private string mainModuleFileName = string.Empty;
+    private bool _isPackaged = false;
 
     [ObservableProperty]
-    private string cpuArchitecture = string.Empty;
+    private bool _isRunningAsAdmin = false;
 
     [ObservableProperty]
-    private bool isPackaged = false;
+    private bool _isRunningAsSystem = false;
 
     [ObservableProperty]
-    private bool isStoreApp = false;
+    private Visibility _visibility = Visibility.Visible;
 
     [ObservableProperty]
-    private bool isRunningAsAdmin = false;
+    private string _activationArgs = string.Empty;
 
     [ObservableProperty]
-    private bool isRunningAsSystem = false;
+    private PackageInfo _packageInfo = new();
 
     [ObservableProperty]
-    private Visibility visibility = Visibility.Visible;
+    private string _identifiedFrameWorkTypes = string.Empty;
 
-    public ObservableCollection<FrameworkType> FrameworkTypes { get; } = [];
+    private List<FrameworkType> FrameworkTypes { get; } = [];
 
     public AppRuntimeInfo()
     {
-        // Note these are in alphabetical order of Name.
         FrameworkTypes.Add(new FrameworkType("Avalonia.Base.dll", "Avalonia"));
         FrameworkTypes.Add(new FrameworkType("DXCore.dll", "DirectX"));
         FrameworkTypes.Add(new FrameworkType("Microsoft.Maui.dll", "Maui"));
         FrameworkTypes.Add(new FrameworkType("MFC", "MFC", false));
         FrameworkTypes.Add(new FrameworkType("Python.exe", "Python"));
+        FrameworkTypes.Add(new FrameworkType("Microsoft.Web.WebView2.Core.dll", "WebView2"));
         FrameworkTypes.Add(new FrameworkType("Microsoft.Windows.SDK.NET.dll", "Windows App SDK"));
         FrameworkTypes.Add(new FrameworkType("System.Windows.Forms.dll", "Windows Forms"));
-        FrameworkTypes.Add(new FrameworkType("Microsoft.WinUI.dll", "WinUI"));
+        FrameworkTypes.Add(new FrameworkType("Microsoft.UI.Xaml.dll", "WinUI 2"));
+        FrameworkTypes.Add(new FrameworkType("Microsoft.UI.Xaml.Controls.dll", "WinUI 3"));
         FrameworkTypes.Add(new FrameworkType("PresentationFramework.dll", "WPF"));
     }
 
-    public void CheckFrameworkTypes(string moduleName)
+    public void CheckFrameworksAndCommandLine(Process process)
```

Comment:
I can't figure out the purpose of this method.  When I see "Check" in a name, I think this would return a bool.  For example, "IsOkay" or something.

Looking at the end I see that `IdentifiedFrameworkTypes` and `ActivationArgs` is assigned values.  If that is the case can you mention the side effect of these variables being assigned a value, or change the name to make the behavior more apparent?

Reply:
Renamed to GetFrameworksAndCommandLine.